backend/wmg/api/v1.py

```diff
@@ -1,58 +1,141 @@
 
-    def build_viz_dots(cell_type_term_ids_: List[str]) -> List[dict]:
-        return [
-            build_viz_dot(cell_type_term_id, me=0.0, pc=0.0, n=0, tpc=0.0) for cell_type_term_id in cell_type_term_ids_
-        ]
+    include_filter_dims = request.get("include_filter_dims", False)
+    response_filter_dims_values = build_filter_dims_values(all_filter_dims_values) if include_filter_dims else {}
 
-    def build_tissues_types(tissue_type_term_ids_: List[str]) -> Dict[str, List]:
-        return dict(
-            [(tissue_type_term_id, build_viz_dots(cell_type_term_ids)) for tissue_type_term_id in tissue_type_term_ids_]
-        )
-
-    def build_genes(gene_term_ids_: List[str]) -> Dict[str, dict]:
-        return dict([(gene_term_id, build_tissues_types(tissue_type_term_ids)) for gene_term_id in gene_term_ids_])
-
-    def build_gene_id_label_mapping(gene_term_ids_) -> List[dict]:
-        return [{gene_term_id: f"{gene_term_id}_label"} for gene_term_id in gene_term_ids_]
-
-    def build_cell_type_id_label_mapping(cell_type_term_ids_) -> List[dict]:
-        return [{cell_type_term_id: f"{cell_type_term_id}_label"} for cell_type_term_id in cell_type_term_ids_]
+    cell_type_term_ids = all_filter_dims_values["cell_type_ontology_term_id"]
 
     return jsonify(
         dict(
-            snapshot_id=DUMMY_SNAPSHOT_UUID,
-            expression_summary=build_genes(gene_term_ids),
+            snapshot_id=snapshot_identifier,
+            expression_summary=build_expression_summary(dot_plot_matrix_df),
             term_id_labels=dict(
-                genes=build_gene_id_label_mapping(gene_term_ids),
-                cell_types=build_cell_type_id_label_mapping(cell_type_term_ids),
+                genes=build_gene_id_label_mapping(criteria.gene_ontology_term_ids),
+                cell_types=build_ontology_term_id_label_mapping(cell_type_term_ids),
             ),
+            filter_dims=response_filter_dims_values,
         )
     )
+
+
+# TODO: It would be reasonable to fetch datasets via a REST API call to Data Portal, rather than coupling to the
+#  portal's db layer, considering that this is the only occasion where the backend.wmg.api package introduces a
+#  dependency on the db. There is no appropriate API call at this time and the lower of making an API call would
+#  have to be considered as well.
+# TODO: This needs a test, but since we are likely to replace it (see above TODO), it is currently not tested
+def fetch_datasets_metadata(dataset_ids: List[str]) -> List[Dict]:
+    # We return a DTO because the db entity can't access its attributes after the db session ends,
+    # and we want to keep session management out of the calling method
+
+    with db_session_manager() as session:
+
+        def get_dataset(dataset_id_):
+            dataset = Dataset.get(session, dataset_id_)
+            if dataset is None:
+                # Handle possible missing dataset due to db state evolving past wmg snapshot
```

Comment:
This is a good call out re data lifecycle. If we run the cube creation batch job weekly there is a high chance things may get out of sync. Will this result in a blank spot/confusing data in the front end display?

Reply:
Actually I think this should be solved by https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell-data-portal/2085 and the corresponding data serving ticket

Reply:
Agreed--that generated data artifact address the problem perfectly. Resolving.